pageserver/src/page_service.rs

```diff
@@ -1326,24 +1330,29 @@ where
                 .for_command(ComputeCommandKind::Basebackup)
                 .inc();
 
-            let (lsn, gzip) = match (params.get(2), params.get(3)) {
-                (None, _) => (None, false),
-                (Some(&"--gzip"), _) => (None, true),
-                (Some(lsn_str), gzip_str_opt) => {
-                    let lsn = Lsn::from_str(lsn_str)
-                        .with_context(|| format!("Failed to parse Lsn from {lsn_str}"))?;
-                    let gzip = match gzip_str_opt {
-                        Some(&"--gzip") => true,
-                        None => false,
-                        Some(third_param) => {
-                            return Err(QueryError::Other(anyhow::anyhow!(
-                                "Parameter in position 3 unknown {third_param}",
-                            )))
+            let mut lsn = None;
+            let mut replica = false;
+            let mut gzip = false;
+            for i in 2..params.len() {
+                if let Some(param) = params.get(i) {
+                    if param.starts_with("--") {
+                        match *param {
+                            "--gzip" => gzip = true,
+                            "--replica" => replica = true,
+                            _ => {
+                                return Err(QueryError::Other(anyhow::anyhow!(
+                                    "Unklnown parameter {param}",
+                                )))
+                            }
                         }
-                    };
-                    (Some(lsn), gzip)
+                    } else {
+                        lsn = Some(
```

Comment:
This allows LSN to be anywhere within the command, instead of enforcing `--flag` must be after LSNs. We can merge it now and I'll refactor the parsing later.

Reply:
could you please also error if lsn is set more than once?

Reply:
Done in 2a95a51a0df6602aa444fd07842e4b347f970ac9